### PR TITLE
Issue120/cp edition view step 4b

### PR DESCRIPTION
**Pull request by richard0000**
# Description
Allow viewing, editing, adding and removing agents on a DSO inside the Configuration Profile edition page.

# Screen Captures
cp-dso-agents

### CODE REVIEW

File: app/javascript/components/dashboard/configuration-profiles/edit/Agents.jsx [hunk 1]
```diff
@@ -0,0 +1,296 @@
...
+import updateCP from "../../../../services/updateCP";
+import ConfirmDialog from "../../../shared/ConfirmDialog";
+
+const Agents = (props) => {
+  const { currentDSOIndex } = props;
+  const [agentsData, setAgentsData] = useState(props.agentsData);
+  const [currentAgentIndex, setCurrentAgentIndex] = useState(0);
+  const configurationProfile = useSelector((state) => state.currentCP);
+  const [agentFullname, setAgentFullname] = useState(
+    agentsData[currentAgentIndex].fullname
```
Comment (excelsior) [inline app/javascript/components/dashboard/configuration-profiles/edit/Agents.jsx:17]: An exception will occur if the `agentsData` array is empty.

File: app/controllers/api/v1/configuration_profiles_controller.rb [hunk 2]
```diff
@@ -22,7 +22,7 @@ class Api::V1::ConfigurationProfilesController < ApplicationController
         :homepage_url,
         :standards_page,
         dso_administrator: %i[fullname email phone github_handle],
-        dso_agents: []
+        dso_agents: [%i[fullname email phone github_handle]]
```
Comment (excelsior) [inline app/controllers/api/v1/configuration_profiles_controller.rb:25]: This array is two-dimensional.

**Review by excelsior — CHANGES_REQUESTED** [review]
The `Agents` component doesn't account for a no data scenario. When a new DSO gets added, its agent list is empty: 

https://github.com/learningtapestry/desm/blob/main/app/javascript/components/dashboard/configuration-profiles/edit/DSOsInfo.jsx#L63

Thus navigating to this component from a newly added DSO will fail. Let's pass an empty agent object when initializing a DSO (or take care of the no data case elsehow).

There's also a small UI issue. When adding more mappers, at some point the UI goes awry (see screenshot). I guess it's not a problem on wider screens, but would be nice to fix it as well.

Screen Shot 2021-10-10 at 23 54 23

File: app/javascript/components/dashboard/configuration-profiles/edit/Agents.jsx (hunk 1 above)
Comment (richard0000) [reply app/javascript/components/dashboard/configuration-profiles/edit/Agents.jsx]: Covered.

**Pull request merged**